alchemy/alchemy.py

```diff
@@ -1262,6 +1541,51 @@ def perturbSystem(cls, system, alchemical_state):
 
         return
 
+    def getEnergyComponents(self, alchemical_state, positions, box_vectors=None, use_all_parameters=True):
+        """
+        Compute potential energy by Force component for the corresponding alchemically-modified system.
+
+        Parameters
+        ----------
+        alchemical_state : AlchemicalState
+            The alchemical state to est the Context to.
+        use_all_parameters : bool, optional, default=False
+            If True, will ensure that all parameters are used or raise an Exception if not.
+
+        """
+        if not hasattr(self, 'alchemically_modified_system_with_force_groups'):
+            # Create deep copy of alchemical system.
+            import copy
+            system = copy.deepcopy(self.alchemically_modified_system)
+            # Separate all forces into separate force groups.
+            assert (system.getNumForces() < 16), "self.alchemically_modified_system has more than 16 force groups; can't compute individual force component energies."
```

Comment:
@jchodera is there a particular reason why you set this to 16? One of the test cases have exactly 16 forces and the openmm docs says that legal values for `.getState()` go from 0 to 31.